addons/l10n_tw_edi_ecpay/models/account_move.py

```diff
@@ -0,0 +1,781 @@
+            formatted_phone = self._reformat_phone_number(self.partner_id.phone)
+            if not re.fullmatch(r'[\d+ ]+', formatted_phone):
+                errors.append(self.env._("Phone number contains invalid characters! Only digits, '+' and spaces are allowed."))
+
+        if self.l10n_tw_edi_is_b2b and not self.partner_id.vat:
+            errors.append(self.env._("Tax id is required for company contact or individual contact under a company."))
+
+        if self.l10n_tw_edi_is_b2b and self.partner_id.vat and (not self.partner_id.vat.isdigit() or len(self.partner_id.vat) != 8):
+            errors.append(self.env._("Tax id is invalid. Should be 8 digits."))
+
+        errors.extend(self._l10n_tw_edi_check_tax_type_on_invoice_lines())
+
+        tax_type, _, is_zero_tax_rate = self._l10n_tw_edi_compute_tax_type_from_invoice_lines()
+
+        if self.l10n_tw_edi_invoice_type == "07" and tax_type not in ["1", "2", "3", "9"]:
+            errors.append(self.env._(
+                "Invoice type 07 must be used with tax type 1, 2, 3 or 9. Please check the tax type on the invoice lines."
+            ))
+
+        if self.l10n_tw_edi_invoice_type == "08" and tax_type not in ["3", "4"]:
+            errors.append(self.env._(
+                "Invoice type 08 must be used with tax type 3 or 4. Please check the tax type on the invoice lines."
+            ))
+
+        if is_zero_tax_rate:
+            if not self.l10n_tw_edi_clearance_mark or not self.l10n_tw_edi_zero_tax_rate_reason:
+                errors.append(self.env._(
+                    "Clearance mark and Zero tax rate reason is required for zero tax rate invoice."
+                ))
+        if errors:
+            if self.website_id:
+                self.message_post(body="Error:\n" + "\n".join(errors))
+            else:
+                raise UserError("Error:\n" + "\n".join(errors))
+
+    def _l10n_tw_edi_prepare_item_list(self, json_data, is_allowance=False):
+        item_list = []
+        sale_amount = 0
+        tax_amount = 0
+        AccountTax = self.env['account.tax']
+        for line in self.invoice_line_ids.filtered(lambda line: line.display_type == "product"):
+
+            base_line = self._prepare_product_base_line_for_taxes_computation(line)
+            if is_allowance and self.reversed_entry_id.currency_id == self.currency_id:
+                base_line['rate'] = self.reversed_entry_id.invoice_currency_rate  # replace the rate by the original invoice's rate
+            AccountTax._add_tax_details_in_base_line(base_line, self.company_id)
+
+            twd_excluded_amount = base_line['tax_details']['raw_total_excluded']
+            twd_included_amount = base_line['tax_details']['raw_total_included']
+
+            tax_type, _, _ = self._l10n_tw_edi_compute_tax_type_from_invoice_lines()
+
+            if self.l10n_tw_edi_is_b2b:
+                item_price = float_round(twd_excluded_amount / line.quantity, precision_rounding=0.01)
```

Comment:
You may also want to do float_repr after rounding if their API are strict about the 2 digits.
float_round only could end up with, E.g. 18.2400000000000001
```suggestion
                item_price = float_repr(float_round(twd_excluded_amount / line.quantity, 2), 2)
```